lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js

```diff
@@ -49,12 +49,23 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
       });
     }
 
+    let debugString;
     const pageHost = new URL(artifacts.URL.finalUrl).host;
     // Filter usages to exclude anchors that are same origin
     // TODO: better extendedInfo for anchors with no href attribute:
     // https://github.com/GoogleChrome/lighthouse/issues/1233
+    // https://github.com/GoogleChrome/lighthouse/issues/1345
     const failingAnchors = artifacts.AnchorsWithNoRelNoopener.usages
-      .filter(anchor => anchor.href === '' || new URL(anchor.href).host !== pageHost)
+      .filter(anchor => {
+        try {
+          return new URL(anchor.href).host !== pageHost;
+        } catch (err) {
+          debugString = 'Lighthouse was unable to determine the destination ' +
+              'of some anchor tags. Remove the href attribute if they are not ' +
```

Comment:
One of the things Brendan discovered in the last PR for this audit was that many SPAs don't include an href attr and use click handlers instead. You catch will include those cases, but maybe we should modify the debug string to say so. E.g. the missing href is intentional in their app, but we're still saying remove it.